src/main/kotlin/pt/up/fe/ni/website/backend/service/AuthService.kt

```diff
@@ -57,26 +59,84 @@ class AuthService(
         return generateAccessToken(account)
     }
 
+    fun generateRecoveryToken(email: String): String? {
+        val account =
+            try {
+                accountService.getAccountByEmail(email)
+            } catch (e: Exception) {
+                logger.error(ErrorMessages.emailNotFound(email))
+                return null
+            }
+        return generateToken(
+            account,
+            Duration.ofMinutes(authConfigProperties.jwtRecoveryExpirationMinutes),
+            usePasswordHash = true
+        )
+    }
+
+    fun confirmRecoveryToken(recoveryToken: String, dto: PasswordRecoveryConfirmDto): Account? {
+        val jwt =
+            try {
+                jwtDecoder.decode(recoveryToken)
+            } catch (e: Exception) {
+                // throw InvalidBearerTokenException(ErrorMessages.invalidRecoveryToken)
+                logger.error(ErrorMessages.invalidRecoveryToken)
```

Comment:
I don't think error makes sense here. logger.errors is usually for server-side or unexpected errors, this is just a malformed request. Imo info or warn makes more sense